### DISCUSSION TITLE
Data Loss Concerns in SeaweedFS EC Configuration

### DISCUSSION THREAD

**Discussion by gigzz** [seaweedfs/seaweedfs#4726] [Q&A]
Hello,

I have set up SeaweedFS Master with the following commands to enable Erasure Coding (EC) and perform related operations:

```
lock
ec.encode -fullPercent=95 -quietFor=1h
ec.rebuild -force
ec.balance -force
volume.deleteEmpty -quietFor=24h -force
volume.balance -force
volume.fix.replication
unlock
```
While these commands provide EC, I have a question regarding the data safety of volumes that are not EC-enabled and the possibility of data loss if one of the hard disks hosting such volumes gets damaged.

As I understand, the EC configuration is intended to enhance data redundancy and protection against disk failures. However, what happens if a volume is not configured with EC, and a disk containing non-EC-enabled volumes becomes damaged? Will the data in those volumes be at risk of permanent loss?

I would like to ensure comprehensive data protection and avoid any potential data loss due to hardware failure. What measures can I take to address this concern and ensure the highest level of data integrity?

Any insights or recommendations on how to handle EC configuration for volumes and protect data in scenarios like these would be greatly appreciated.

Thank you for your assistance!

Best regards

**Comment by chrislusf** [comment]
usually you would want to write data with replication enabled.
